x11-drivers/nvidia-drivers/nvidia-drivers-465.19.01.ebuild

```diff
@@ -0,0 +1,399 @@
+		doins "${FILESDIR}"/nvidia-blacklist-nouveau.conf
+		doins "${FILESDIR}"/nvidia-rmmod.conf
+
+		# used for gpu verification with binpkgs (not kept)
+		insinto /usr/share/nvidia
+		doins supported-gpus/supported-gpus.json
+	fi
+
+	if use X; then
+		exeinto /usr/$(get_libdir)/xorg/modules/drivers
+		doexe nvidia_drv.so
+
+		exeinto /usr/$(get_libdir)/xorg/modules/extensions
+		newexe libglxserver_nvidia.so{.${PV},}
+
+		insinto /usr/share/X11/xorg.conf.d
+		newins {,50-}nvidia-drm-outputclass.conf
+
+		# vulkan icd uses libGLX_nvidia.so and so requires X
+		insinto /usr/share/vulkan/icd.d
+		doins nvidia_icd.json
+		insinto /usr/share/vulkan/implicit_layer.d
+		doins nvidia_layers.json
+	fi
+
+	insinto /usr/share/glvnd/egl_vendor.d
+	doins 10_nvidia.json
+
+	insinto /etc/OpenCL/vendors
+	doins nvidia.icd
+
+	insinto /etc/nvidia
+	newins nvidia-application-profiles{-${PV},}-rc
+
+	# install built helpers
+	nvidia-drivers_make modprobe install
+	# allow video group to load mods and create devs (bug #505092)
+	fowners root:video /usr/bin/nvidia-modprobe
+	fperms 4710 /usr/bin/nvidia-modprobe
+
+	nvidia-drivers_make persistenced install
+	newconfd "${FILESDIR}/nvidia-persistenced.confd" nvidia-persistenced
+	newinitd "${FILESDIR}/nvidia-persistenced.initd" nvidia-persistenced
+	systemd_dounit nvidia-persistenced.service
+
+	use X && nvidia-drivers_make xconfig install
+
+	if use tools; then
+		nvidia-drivers_make settings install
+		doicon nvidia-settings/doc/nvidia-settings.png
+		domenu nvidia-settings/doc/nvidia-settings.desktop
+
+		insinto /usr/share/nvidia
+		newins nvidia-application-profiles{-${PV},}-key-documentation
+
+		exeinto /etc/X11/xinit/xinitrc.d
+		newexe "${FILESDIR}"/95-nvidia-settings-r1 95-nvidia-settings
+	fi
+
+	if use static-libs; then
+		dolib.a nvidia-settings/src/out/libXNVCtrl.a
+
+		insinto /usr/include/NVCtrl
+		doins nvidia-settings/src/libXNVCtrl/NVCtrl{Lib,}.h
+	fi
+
+	# install prebuilt-only helpers
+	exeinto /opt/bin
+
+	doexe nvidia-cuda-mps-control
+	doman nvidia-cuda-mps-control.1
+	doexe nvidia-cuda-mps-server
+
+	doexe nvidia-debugdump
+	dobin nvidia-bug-report.sh
+
+	doexe nvidia-smi
+	doman nvidia-smi.1
+
+	# install prebuilt-only libraries
+	multilib_foreach_abi nvidia-drivers_libs_install
+
+	# systemd sleep services
+	exeinto /lib/systemd/system-sleep
+	doexe systemd/system-sleep/nvidia
+	dobin systemd/nvidia-sleep.sh
+	systemd_dounit systemd/system/nvidia-{hibernate,resume,suspend}.service
+
+	einstalldocs
+}
+
+pkg_preinst() {
+	use driver || return
+	linux-mod_pkg_preinst
+
+	# set video group id based on live system (bug #491414)
+	local g=$(getent group video | cut -d: -f3)
+	[[ ${g} ]] || die "Failed to determine video group id"
+	sed "s/PACKAGE/${PF}/;s/VIDEOGID/${g}/" \
+		-i "${ED}"/etc/modprobe.d/nvidia.conf || die
+
+	# try to find driver mismatches using temporary supported-gpus.json
+	for g in $(grep -l 0x10de /sys/bus/pci/devices/*/vendor 2>/dev/null); do
```

Comment:
I suppose we could/should explicitly BDEPEND on grep too.

Reply:
Ah yes, unlike sed/patch/find it's not guaranteed by pms during build.

Reply:
Oh wait, actually this should be RDEPEND, this is pkg_preinst

Reply:
Or rather IDEPEND if we had it... cross-compiling nvidia-drivers is already rather fickle either way.